src/pages/user/institution/InstitutionCard.tsx

```diff
@@ -34,58 +25,94 @@ const StyledTextContainer = styled.div`
-    }
-  };
-
-  const handleEditInstitution = async () => {
-    onEdit();
-    setFieldValue(FormikInstitutionUnitFieldNames.ID, unit.id);
-    setFieldValue(FormikInstitutionUnitFieldNames.NAME, unit.name);
-    setFieldValue(FormikInstitutionUnitFieldNames.SUBUNITS, unit.subunits);
-    setFieldValue(FormikInstitutionUnitFieldNames.UNIT, unit.unit);
-    setFieldValue(FormikInstitutionUnitFieldNames.EDIT_ID, organizationUnitId);
-  };
+  const [unit, setUnit] = useState<RecursiveInstitutionUnit>();
+  const [isLoadingUnit, setIsLoadingUnit] = useState(false);
+
+  useEffect(() => {
+    const fetchDepartment = async () => {
+      setIsLoadingUnit(true);
+      const isSubunit = orgunitId.includes('.');
+      const unitUri = isSubunit
+        ? `https://api.cristin.no/v2/units/${orgunitId}`
+        : `https://api.cristin.no/v2/institutions/${orgunitId}`;
+      const response = await getDepartment(unitUri);
+      if (response?.error) {
+        dispatch(setNotification(response.error, NotificationVariant.Error));
+      } else {
+        const unit = JSON.parse(response.json);
+        if (!isSubunit) {
+          // Remove subunits from institution, since we only care about top-level in this case
+          delete unit.subunits;
+        }
+        setUnit(unit);
+      }
+      setIsLoadingUnit(false);
+    };
+
+    fetchDepartment();
+  }, [dispatch, orgunitId]);
+
+  // const organizationUnitId = unit?.subunits && unit.subunits.length > 0 ? unit.subunits.slice(-1)[0].id : unit.id;
+  // const { setFieldValue }: FormikProps<FormikInstitutionUnit> = useFormikContext();
+
+  // const handleRemoveInstitution = async () => {
+  //   if (!authority || !orgunitId) {
+  //     return;
+  //   }
+  //   const updatedAuthority = await removeQualifierIdFromAuthority(
+  //     authority.systemControlNumber,
+  //     AuthorityQualifiers.ORGUNIT_ID,
+  //     orgunitId
+  //   );
+  //   if (updatedAuthority.error) {
+  //     dispatch(setNotification(updatedAuthority.error, NotificationVariant.Error));
+  //   } else if (updatedAuthority) {
+  //     dispatch(setAuthorityData(updatedAuthority));
+  //     dispatch(setNotification(t('feedback:success.delete_identifier')));
+  //   }
+  // };
+
+  // const handleEditInstitution = async () => {
+  //   // onEdit();
+  //   setFieldValue(FormikInstitutionUnitFieldNames.ID, unit.id);
+  //   setFieldValue(FormikInstitutionUnitFieldNames.NAME, unit.name);
+  //   setFieldValue(FormikInstitutionUnitFieldNames.SUBUNITS, unit.subunits);
+  //   setFieldValue(FormikInstitutionUnitFieldNames.UNIT, unit.unit);
+  //   setFieldValue(FormikInstitutionUnitFieldNames.EDIT_ID, organizationUnitId);
+  // };
 
   return (
-    <StyledSelectedInstitution data-testid="institution-presentation">
-      <StyledTextContainer>
+    <StyledCard data-testid="institution-presentation">
+      {isLoadingUnit ? (
+        <Progress />
+      ) : (
+        <StyledTextContainer>
```

Comment:
dette burde dras ut til egen komponent, sånn at den kan brukes for contributors också

Reply:
Tar det i egen sak https://unit.atlassian.net/browse/NP-845